examples/CIB/ex1/main.cpp

```diff
@@ -399,6 +495,83 @@ main(int argc, char* argv[])
             pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
             pout << "\n";
 
+            // Fill ghost cells of pressure and velocity to compute hydrodynamic forces.
```

Comment:
If you are worried about efficiency, one thing that you could do would be to "register" velocity and pressure Variables with the class, and to have the class setup the necessary patch data indices as needed at that time.